dockerplugin.py

```diff
@@ -480,7 +567,9 @@ def init_callback(self):
     def read_callback(self):
         try:
             containers = [c for c in self.client.containers()
-                          if c['Status'].startswith('Up')]
+                          if c['Status'].startswith('Up')
+                          and not self.is_excluded(c)]
```

Comment:
I wouldn't check for the exclusion here, because that means you're doing it every single time on every `read_callback`. Get all the containers here, and only do the exclusion check line 599 if the container is not in `self.stats`. This way, if you already have a `ContainerStats` object, you know you are not excluding this container, but you also don't have to recheck if it's excluded every time because that's not gonna change.